### ISSUE TITLE
Spreading a get accessor property results in false positive "overwritten" error

### ISSUE THREAD

**Issue by Dionysusnu** [microsoft/TypeScript#47961]
# Bug Report

### 🔎 Search Terms

spread, get, accessor, class

### 🕗 Version & Regression Information

- This changed between versions 3.8.3 and 3.9.7

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play#code/MYGwhgzhAECC0G8CwAoa7oHMCmAXaARmAE4AUAlIqhjdMXgK7EB20ARAPIBu2xA7sQCWuXNlYEAnljyjibANzUMAX1SqUqYAHtmEfADMtW6AF5ozbHzgVNOvdDCmqaDAHpX0AGJgQEbAFoABy0IYUEeaF5iLWJCbGAwBj9oAAMiYhToPkEQEHMtfAJsaC0efiERMUIpRxx8MGBgbCgYpXR0gC52AHVIfPxS3gFhUWY2ABo26AA6WcMtSZR1d1T0zMEYbWJ6YFwQKWYC6EFWNJSAchhcCUDscbiEpOLhS+h9aIBbB2l6xuaIVoobS6QpOZAudCzabzRbqVArEBaTAwFIINjpNhdNi9GCHAZlYaVMbKFK2XRaEDYaaIzCkMDkeEeGkohAkskAynUpGkAjkIA)

### 💻 Code

```ts
class A {
    get bar() {
        return "Overwritten by getter";
    }
}

const foo = new A()
const a = {
    // False-positive error because `bar` will not be overwritten by a get accessor
    bar: "Was not overwritten",
    ...foo,
}
// `bar` is correctly not in `b`'s type, because it's from a get accessor
const b = {
    ...foo,
}

// logs `{"bar": "Was not overwritten"}`
console.log(a)
// logs `{}`
console.log(b)
```

### 🙁 Actual behavior

An error is thrown on line 10, reporting that `bar` is overwritten. However, get accessors do not overwrite properties during spread behaviour.

### 🙂 Expected behavior

The error should not be reported, because `bar` is not actually overwritten. This is already recognised, as the type of `b` shows the `bar` property is excluded when not specified in the literal.

**Comment by RyanCavanaugh** [comment]
See #9726 - enumerability is not tracked

**Comment by Dionysusnu** [comment]
It's kind of a separate bug though, isn't it? The type of `b` shows clearly that the enumerability is tracked in _some_ way. It seems odd that that would then not be used for the overwriting diagnostic.
